editor/animation_track_editor.cpp

```diff
@@ -4989,7 +4987,9 @@ void AnimationTrackEditor::_update_step(double p_new_step) {
 
 	EditorUndoRedoManager *undo_redo = EditorUndoRedoManager::get_singleton();
 	undo_redo->create_action(TTR("Change Animation Step"));
-	float step_value = p_new_step;
+	// To ensure that the conversion results are consistent between serialization and load, the value is snapped with 0.0625 to be a rational number.
+	// step_value must also be less than or equal to 1000 to ensure that no error accumulates due to interactions with retrieving values from inner range.
+	double step_value = MIN(1000.0f, Math::snapped(p_new_step, FACTOR));
```

Comment:
```suggestion
	double step_value = MIN(1000.0, Math::snapped(p_new_step, FACTOR));
```

Reply:
This still needs to be resolved, you're mixing types here

Reply:
fixed, I must have not committed this change before I last rebased things...thought I did but it must have slipped under the radar